script.js

```diff
@@ -1 +1,402 @@
-console.log('yay!');
+// Global setup ----------------------------------------------------
+let board = [];
+let firstCard = null;
+const boardSize = 4; // has to be an even number
+const matchesNeededToWin = (boardSize * boardSize) / 2;
+let numOfMatches = 0;
+let deck;
+let playerName;
+let secondsLeftTillGameOver = 180;
+let numOfGames = 1;
+const loseMessage = 'Sorry but the time ran out! Click on a square to try again!';
+let timerId; // to store ID of startMatchTimer
+
+let gameStatus = 'playGame';
+
+// to store the square elements that the player clicked on this round
+let firstClickedSquare;
+let secondClickedSquare;
+
+// player cant click on squares at start of game
+// he or she must submit their name first
+let canClick = false;
+
+const gameInfo = document.createElement('div');
+
+// container to display score message
+const scoreMessage = document.createElement('div');
+
+const playerNameInput = document.createElement('input');
+const playerNameButton = document.createElement('button');
+// Container for player name input and button
+const playerNameInputContainer = document.createElement('div');
+
+// container to display match's countdown timer
+const matchCountdownTimer = document.createElement('div');
```

Comment:
Consider initialising these DOM elements inside the `gameInit` function. I understand we want to declare the vars as globals and here is a good place, but the initialisation can happen inside `gameInit` so that all DOM initialisation happens in the same place for clarity.